Build failed using JDK 17

## Bug Report

### Which version of ShardingSphere did you use?
master

### Expected behavior

### Actual behavior

Java version:
```
➜  shardingsphere git:(master) ✗ $JAVA_HOME/bin/java -version
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true
java version "17" 2021-09-14 LTS
Java(TM) SE Runtime Environment (build 17+35-LTS-2724)
Java HotSpot(TM) 64-Bit Server VM (build 17+35-LTS-2724, mixed mode, sharing)
➜  shardingsphere git:(master) ✗ $JAVA_HOME/bin/javac -version
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true
javac 17
```

Executing `./mvnw clean install`
```
➜  shardingsphere git:(master) ✗ ./mvnw clean install
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: osx
[INFO] os.detected.arch: x86_64
[INFO] os.detected.version: 11.4
[INFO] os.detected.version.major: 11
[INFO] os.detected.version.minor: 4
[INFO] os.detected.classifier: osx-x86_64
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] shardingsphere                                                     [pom]
[INFO] shardingsphere-spi                                                 [jar]
[INFO] shardingsphere-sql-parser                                          [pom]
[INFO] shardingsphere-sql-parser-spi                                      [jar]
[INFO] shardingsphere-sql-parser-statement                                [jar]
[INFO] shardingsphere-sql-parser-engine                                   [jar]
[INFO] shardingsphere-distsql-parser                                      [pom]
[INFO] shardingsphere-distsql-parser-statement                            [jar]
[INFO] shardingsphere-distsql-parser-engine                               [jar]
[INFO] shardingsphere-features                                            [pom]
[INFO] shardingsphere-sharding                                            [pom]
[INFO] shardingsphere-sharding-distsql                                    [pom]
[INFO] shardingsphere-sharding-distsql-statement                          [jar]
[INFO] shardingsphere-sharding-distsql-parser                             [jar]
[INFO] shardingsphere-readwrite-splitting                                 [pom]
[INFO] shardingsphere-readwrite-splitting-distsql                         [pom]
[INFO] shardingsphere-readwrite-splitting-distsql-statement               [jar]
[INFO] shardingsphere-readwrite-splitting-distsql-parser                  [jar]
[INFO] shardingsphere-db-discovery                                        [pom]
[INFO] shardingsphere-db-discovery-distsql                                [pom]
[INFO] shardingsphere-db-discovery-distsql-statement                      [jar]
[INFO] shardingsphere-db-discovery-distsql-parser                         [jar]
[INFO] shardingsphere-encrypt                                             [pom]
[INFO] shardingsphere-encrypt-distsql                                     [pom]
[INFO] shardingsphere-encrypt-distsql-statement                           [jar]
[INFO] shardingsphere-encrypt-distsql-parser                              [jar]
[INFO] shardingsphere-scaling                                             [pom]
[INFO] shardingsphere-scaling-distsql                                     [pom]
[INFO] shardingsphere-scaling-distsql-statement                           [jar]
[INFO] shardingsphere-scaling-distsql-parser                              [jar]
[INFO] shardingsphere-shadow                                              [pom]
[INFO] shardingsphere-shadow-distsql                                      [pom]
[INFO] shardingsphere-shadow-distsql-statement                            [jar]
[INFO] shardingsphere-shadow-distsql-parser                               [jar]
[INFO] shardingsphere-test                                                [pom]
[INFO] shardingsphere-parser-test                                         [jar]
[INFO] shardingsphere-sql-parser-dialect                                  [pom]
[INFO] shardingsphere-sql-parser-sql92                                    [jar]
[INFO] shardingsphere-sql-parser-mysql                                    [jar]
[INFO] shardingsphere-sql-parser-postgresql                               [jar]
[INFO] shardingsphere-sql-parser-opengauss                                [jar]
[INFO] shardingsphere-sql-parser-oracle                                   [jar]
[INFO] shardingsphere-sql-parser-sqlserver                                [jar]
[INFO] shardingsphere-db-protocol                                         [pom]
[INFO] shardingsphere-db-protocol-core                                    [jar]
[INFO] shardingsphere-db-protocol-mysql                                   [jar]
[INFO] shardingsphere-db-protocol-postgresql                              [jar]
[INFO] shardingsphere-db-protocol-opengauss                               [jar]
[INFO] shardingsphere-test-common                                         [jar]
[INFO] shardingsphere-infra                                               [pom]
[INFO] shardingsphere-infra-common                                        [jar]
[INFO] shardingsphere-infra-parser                                        [jar]
[INFO] shardingsphere-infra-binder                                        [jar]
[INFO] shardingsphere-infra-route                                         [jar]
[INFO] shardingsphere-infra-rewrite                                       [jar]
[INFO] shardingsphere-infra-optimize                                      [jar]
[INFO] shardingsphere-infra-executor                                      [jar]
[INFO] shardingsphere-infra-merge                                         [jar]
[INFO] shardingsphere-kernel                                              [pom]
[INFO] shardingsphere-authority                                           [pom]
[INFO] shardingsphere-authority-api                                       [jar]
[INFO] shardingsphere-authority-core                                      [jar]
[INFO] shardingsphere-transaction                                         [pom]
[INFO] shardingsphere-transaction-api                                     [jar]
[INFO] shardingsphere-transaction-core                                    [jar]
[INFO] shardingsphere-infra-context                                       [jar]
[INFO] shardingsphere-infra-datetime                                      [pom]
[INFO] shardingsphere-infra-datetime-spi                                  [jar]
[INFO] shardingsphere-infra-datetime-type                                 [pom]
[INFO] shardingsphere-system-datetime                                     [jar]
[INFO] shardingsphere-sharding-api                                        [jar]
[INFO] shardingsphere-sharding-distsql-handler                            [jar]
[INFO] shardingsphere-single-table                                        [pom]
[INFO] shardingsphere-single-table-api                                    [jar]
[INFO] shardingsphere-single-table-core                                   [jar]
[INFO] shardingsphere-sharding-core                                       [jar]
[INFO] shardingsphere-database-datetime                                   [jar]
[INFO] shardingsphere-mode                                                [pom]
[INFO] shardingsphere-mode-core                                           [jar]
[INFO] shardingsphere-mode-type                                           [pom]
[INFO] shardingsphere-memory-mode                                         [pom]
[INFO] shardingsphere-memory-mode-core                                    [jar]
[INFO] shardingsphere-standalone-mode                                     [pom]
[INFO] shardingsphere-standalone-mode-repository                          [pom]
[INFO] shardingsphere-standalone-mode-repository-api                      [jar]
[INFO] shardingsphere-standalone-mode-core                                [jar]
[INFO] shardingsphere-standalone-mode-repository-provider                 [pom]
[INFO] shardingsphere-standalone-mode-repository-file                     [jar]
[INFO] shardingsphere-cluster-mode                                        [pom]
[INFO] shardingsphere-cluster-mode-repository                             [pom]
[INFO] shardingsphere-cluster-mode-repository-api                         [jar]
[INFO] shardingsphere-cluster-mode-core                                   [jar]
[INFO] shardingsphere-cluster-mode-repository-provider                    [pom]
[INFO] shardingsphere-cluster-mode-repository-zookeeper-curator           [jar]
[INFO] shardingsphere-cluster-mode-repository-etcd                        [jar]
[INFO] shardingsphere-transaction-type                                    [pom]
[INFO] shardingsphere-transaction-xa                                      [pom]
[INFO] shardingsphere-transaction-xa-spi                                  [jar]
[INFO] shardingsphere-transaction-xa-provider                             [pom]
[INFO] shardingsphere-transaction-xa-atomikos                             [jar]
[INFO] shardingsphere-transaction-xa-core                                 [jar]
[INFO] shardingsphere-transaction-xa-narayana                             [jar]
[INFO] shardingsphere-transaction-xa-bitronix                             [jar]
[INFO] shardingsphere-transaction-base                                    [pom]
[INFO] shardingsphere-transaction-base-seata-at                           [jar]
[INFO] shardingsphere-readwrite-splitting-api                             [jar]
[INFO] shardingsphere-readwrite-splitting-distsql-handler                 [jar]
[INFO] shardingsphere-readwrite-splitting-core                            [jar]
[INFO] shardingsphere-db-discovery-api                                    [jar]
[INFO] shardingsphere-db-discovery-distsql-handler                        [jar]
[INFO] shardingsphere-db-discovery-provider                               [pom]
[INFO] shardingsphere-db-discovery-mgr                                    [jar]
[INFO] shardingsphere-db-discovery-core                                   [jar]
[INFO] shardingsphere-encrypt-api                                         [jar]
[INFO] shardingsphere-encrypt-distsql-handler                             [jar]
[INFO] shardingsphere-encrypt-core                                        [jar]
[INFO] shardingsphere-shadow-api                                          [jar]
[INFO] shardingsphere-shadow-distsql-handler                              [jar]
[INFO] shardingsphere-shadow-core                                         [jar]
[INFO] shardingsphere-jdbc                                                [pom]
[INFO] shardingsphere-jdbc-core                                           [jar]
[INFO] shardingsphere-jdbc-spring                                         [pom]
[INFO] shardingsphere-jdbc-spring-infra                                   [pom]
[INFO] shardingsphere-jdbc-spring-boot-starter-infra                      [jar]
[INFO] shardingsphere-jdbc-spring-namespace-infra                         [jar]
[INFO] shardingsphere-sharding-spring                                     [pom]
[INFO] shardingsphere-sharding-spring-boot-starter                        [jar]
[INFO] shardingsphere-shadow-spring                                       [pom]
[INFO] shardingsphere-shadow-spring-boot-starter                          [jar]
[INFO] shardingsphere-encrypt-spring                                      [pom]
[INFO] shardingsphere-encrypt-spring-boot-starter                         [jar]
[INFO] shardingsphere-readwrite-splitting-spring                          [pom]
[INFO] shardingsphere-readwrite-splitting-spring-boot-starter             [jar]
[INFO] shardingsphere-jdbc-transaction-spring                             [jar]
[INFO] shardingsphere-jdbc-core-spring                                    [pom]
[INFO] shardingsphere-jdbc-core-spring-boot-starter                       [jar]
[INFO] shardingsphere-sharding-spring-namespace                           [jar]
[INFO] shardingsphere-readwrite-splitting-spring-namespace                [jar]
[INFO] shardingsphere-encrypt-spring-namespace                            [jar]
[INFO] shardingsphere-shadow-spring-namespace                             [jar]
[INFO] shardingsphere-jdbc-core-spring-namespace                          [jar]
[INFO] shardingsphere-proxy                                               [pom]
[INFO] shardingsphere-proxy-common                                        [jar]
[INFO] shardingsphere-scaling-core                                        [jar]
[INFO] shardingsphere-scaling-distsql-handler                             [jar]
[INFO] shardingsphere-scaling-dialect                                     [pom]
[INFO] shardingsphere-scaling-mysql                                       [jar]
[INFO] shardingsphere-scaling-postgresql                                  [jar]
[INFO] shardingsphere-proxy-backend                                       [jar]
[INFO] shardingsphere-proxy-frontend                                      [pom]
[INFO] shardingsphere-proxy-frontend-spi                                  [jar]
[INFO] shardingsphere-proxy-frontend-core                                 [jar]
[INFO] shardingsphere-proxy-frontend-mysql                                [jar]
[INFO] shardingsphere-proxy-frontend-postgresql                           [jar]
[INFO] shardingsphere-proxy-frontend-opengauss                            [jar]
[INFO] shardingsphere-scaling-opengauss                                   [jar]
[INFO] shardingsphere-proxy-bootstrap                                     [jar]
[INFO] shardingsphere-db-discovery-spring                                 [pom]
[INFO] shardingsphere-db-discovery-spring-namespace                       [jar]
[INFO] shardingsphere-db-discovery-spring-boot-starter                    [jar]
[INFO] shardingsphere-agent                                               [pom]
[INFO] shardingsphere-agent-api                                           [jar]
[INFO] shardingsphere-agent-core                                          [jar]
[INFO] shardingsphere-agent-bootstrap                                     [jar]
[INFO] shardingsphere-agent-plugins                                       [pom]
[INFO] shardingsphere-agent-plugin-metrics                                [pom]
[INFO] shardingsphere-agent-metrics-api                                   [jar]
[INFO] shardingsphere-agent-metrics-prometheus                            [jar]
[INFO] shardingsphere-agent-plugin-tracing                                [pom]
[INFO] shardingsphere-agent-tracing-test                                  [jar]
[INFO] shardingsphere-agent-tracing-jaeger                                [jar]
[INFO] shardingsphere-agent-tracing-zipkin                                [jar]
[INFO] shardingsphere-agent-tracing-opentracing                           [jar]
[INFO] shardingsphere-agent-tracing-opentelemetry                         [jar]
[INFO] shardingsphere-agent-plugin-logging                                [pom]
[INFO] shardingsphere-agent-logging-base                                  [jar]
[INFO] shardingsphere-agent-distribution                                  [pom]
[INFO] shardingsphere-scaling-bootstrap                                   [jar]
[INFO] shardingsphere-integration-test                                    [pom]
[INFO] shardingsphere-integration-test-fixture                            [jar]
[INFO] shardingsphere-integration-test-suite                              [jar]
[INFO] shardingsphere-integration-driver-test                             [jar]
[INFO] shardingsphere-integration-agent-test                              [pom]
[INFO] shardingsphere-integration-agent-test-plugins                      [pom]
[INFO] shardingsphere-integration-agent-test-common                       [jar]
[INFO] shardingsphere-integration-agent-test-metrics                      [jar]
[INFO] shardingsphere-integration-agent-test-zipkin                       [jar]
[INFO] shardingsphere-integration-agent-test-jaeger                       [jar]
[INFO] shardingsphere-integration-agent-test-opentelemetry                [jar]
[INFO] shardingsphere-integration-scaling-test                            [pom]
[INFO] shardingsphere-integration-scaling-test-mysql                      [jar]
[INFO] shardingsphere-rewrite-test                                        [jar]
[INFO] shardingsphere-distribution                                        [pom]
[INFO] shardingsphere-src-distribution                                    [pom]
[INFO] shardingsphere-jdbc-distribution                                   [pom]
[INFO] shardingsphere-proxy-distribution                                  [pom]
[INFO] shardingsphere-scaling-distribution                                [pom]
[INFO] 
[INFO] --------------< org.apache.shardingsphere:shardingsphere >--------------
[INFO] Building shardingsphere 5.0.0-RC1-SNAPSHOT                       [1/197]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ shardingsphere ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4:enforce (enforce-maven-version) @ shardingsphere ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4:enforce (enforce-banned-dependencies) @ shardingsphere ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for shardingsphere 5.0.0-RC1-SNAPSHOT:
[INFO] 
[INFO] shardingsphere ..................................... FAILURE [  0.475 s]
[INFO] shardingsphere-spi ................................. SKIPPED
[INFO] shardingsphere-sql-parser .......................... SKIPPED
[INFO] shardingsphere-sql-parser-spi ...................... SKIPPED
[INFO] shardingsphere-sql-parser-statement ................ SKIPPED
[INFO] shardingsphere-sql-parser-engine ................... SKIPPED
[INFO] shardingsphere-distsql-parser ...................... SKIPPED
[INFO] shardingsphere-distsql-parser-statement ............ SKIPPED
[INFO] shardingsphere-distsql-parser-engine ............... SKIPPED
[INFO] shardingsphere-features ............................ SKIPPED
[INFO] shardingsphere-sharding ............................ SKIPPED
[INFO] shardingsphere-sharding-distsql .................... SKIPPED
[INFO] shardingsphere-sharding-distsql-statement .......... SKIPPED
[INFO] shardingsphere-sharding-distsql-parser ............. SKIPPED
[INFO] shardingsphere-readwrite-splitting ................. SKIPPED
[INFO] shardingsphere-readwrite-splitting-distsql ......... SKIPPED
[INFO] shardingsphere-readwrite-splitting-distsql-statement SKIPPED
[INFO] shardingsphere-readwrite-splitting-distsql-parser .. SKIPPED
[INFO] shardingsphere-db-discovery ........................ SKIPPED
[INFO] shardingsphere-db-discovery-distsql ................ SKIPPED
[INFO] shardingsphere-db-discovery-distsql-statement ...... SKIPPED
[INFO] shardingsphere-db-discovery-distsql-parser ......... SKIPPED
[INFO] shardingsphere-encrypt ............................. SKIPPED
[INFO] shardingsphere-encrypt-distsql ..................... SKIPPED
[INFO] shardingsphere-encrypt-distsql-statement ........... SKIPPED
[INFO] shardingsphere-encrypt-distsql-parser .............. SKIPPED
[INFO] shardingsphere-scaling ............................. SKIPPED
[INFO] shardingsphere-scaling-distsql ..................... SKIPPED
[INFO] shardingsphere-scaling-distsql-statement ........... SKIPPED
[INFO] shardingsphere-scaling-distsql-parser .............. SKIPPED
[INFO] shardingsphere-shadow .............................. SKIPPED
[INFO] shardingsphere-shadow-distsql ...................... SKIPPED
[INFO] shardingsphere-shadow-distsql-statement ............ SKIPPED
[INFO] shardingsphere-shadow-distsql-parser ............... SKIPPED
[INFO] shardingsphere-test ................................ SKIPPED
[INFO] shardingsphere-parser-test ......................... SKIPPED
[INFO] shardingsphere-sql-parser-dialect .................. SKIPPED
[INFO] shardingsphere-sql-parser-sql92 .................... SKIPPED
[INFO] shardingsphere-sql-parser-mysql .................... SKIPPED
[INFO] shardingsphere-sql-parser-postgresql ............... SKIPPED
[INFO] shardingsphere-sql-parser-opengauss ................ SKIPPED
[INFO] shardingsphere-sql-parser-oracle ................... SKIPPED
[INFO] shardingsphere-sql-parser-sqlserver ................ SKIPPED
[INFO] shardingsphere-db-protocol ......................... SKIPPED
[INFO] shardingsphere-db-protocol-core .................... SKIPPED
[INFO] shardingsphere-db-protocol-mysql ................... SKIPPED
[INFO] shardingsphere-db-protocol-postgresql .............. SKIPPED
[INFO] shardingsphere-db-protocol-opengauss ............... SKIPPED
[INFO] shardingsphere-test-common ......................... SKIPPED
[INFO] shardingsphere-infra ............................... SKIPPED
[INFO] shardingsphere-infra-common ........................ SKIPPED
[INFO] shardingsphere-infra-parser ........................ SKIPPED
[INFO] shardingsphere-infra-binder ........................ SKIPPED
[INFO] shardingsphere-infra-route ......................... SKIPPED
[INFO] shardingsphere-infra-rewrite ....................... SKIPPED
[INFO] shardingsphere-infra-optimize ...................... SKIPPED
[INFO] shardingsphere-infra-executor ...................... SKIPPED
[INFO] shardingsphere-infra-merge ......................... SKIPPED
[INFO] shardingsphere-kernel .............................. SKIPPED
[INFO] shardingsphere-authority ........................... SKIPPED
[INFO] shardingsphere-authority-api ....................... SKIPPED
[INFO] shardingsphere-authority-core ...................... SKIPPED
[INFO] shardingsphere-transaction ......................... SKIPPED
[INFO] shardingsphere-transaction-api ..................... SKIPPED
[INFO] shardingsphere-transaction-core .................... SKIPPED
[INFO] shardingsphere-infra-context ....................... SKIPPED
[INFO] shardingsphere-infra-datetime ...................... SKIPPED
[INFO] shardingsphere-infra-datetime-spi .................. SKIPPED
[INFO] shardingsphere-infra-datetime-type ................. SKIPPED
[INFO] shardingsphere-system-datetime ..................... SKIPPED
[INFO] shardingsphere-sharding-api ........................ SKIPPED
[INFO] shardingsphere-sharding-distsql-handler ............ SKIPPED
[INFO] shardingsphere-single-table ........................ SKIPPED
[INFO] shardingsphere-single-table-api .................... SKIPPED
[INFO] shardingsphere-single-table-core ................... SKIPPED
[INFO] shardingsphere-sharding-core ....................... SKIPPED
[INFO] shardingsphere-database-datetime ................... SKIPPED
[INFO] shardingsphere-mode ................................ SKIPPED
[INFO] shardingsphere-mode-core ........................... SKIPPED
[INFO] shardingsphere-mode-type ........................... SKIPPED
[INFO] shardingsphere-memory-mode ......................... SKIPPED
[INFO] shardingsphere-memory-mode-core .................... SKIPPED
[INFO] shardingsphere-standalone-mode ..................... SKIPPED
[INFO] shardingsphere-standalone-mode-repository .......... SKIPPED
[INFO] shardingsphere-standalone-mode-repository-api ...... SKIPPED
[INFO] shardingsphere-standalone-mode-core ................ SKIPPED
[INFO] shardingsphere-standalone-mode-repository-provider . SKIPPED
[INFO] shardingsphere-standalone-mode-repository-file ..... SKIPPED
[INFO] shardingsphere-cluster-mode ........................ SKIPPED
[INFO] shardingsphere-cluster-mode-repository ............. SKIPPED
[INFO] shardingsphere-cluster-mode-repository-api ......... SKIPPED
[INFO] shardingsphere-cluster-mode-core ................... SKIPPED
[INFO] shardingsphere-cluster-mode-repository-provider .... SKIPPED
[INFO] shardingsphere-cluster-mode-repository-zookeeper-curator SKIPPED
[INFO] shardingsphere-cluster-mode-repository-etcd ........ SKIPPED
[INFO] shardingsphere-transaction-type .................... SKIPPED
[INFO] shardingsphere-transaction-xa ...................... SKIPPED
[INFO] shardingsphere-transaction-xa-spi .................. SKIPPED
[INFO] shardingsphere-transaction-xa-provider ............. SKIPPED
[INFO] shardingsphere-transaction-xa-atomikos ............. SKIPPED
[INFO] shardingsphere-transaction-xa-core ................. SKIPPED
[INFO] shardingsphere-transaction-xa-narayana ............. SKIPPED
[INFO] shardingsphere-transaction-xa-bitronix ............. SKIPPED
[INFO] shardingsphere-transaction-base .................... SKIPPED
[INFO] shardingsphere-transaction-base-seata-at ........... SKIPPED
[INFO] shardingsphere-readwrite-splitting-api ............. SKIPPED
[INFO] shardingsphere-readwrite-splitting-distsql-handler . SKIPPED
[INFO] shardingsphere-readwrite-splitting-core ............ SKIPPED
[INFO] shardingsphere-db-discovery-api .................... SKIPPED
[INFO] shardingsphere-db-discovery-distsql-handler ........ SKIPPED
[INFO] shardingsphere-db-discovery-provider ............... SKIPPED
[INFO] shardingsphere-db-discovery-mgr .................... SKIPPED
[INFO] shardingsphere-db-discovery-core ................... SKIPPED
[INFO] shardingsphere-encrypt-api ......................... SKIPPED
[INFO] shardingsphere-encrypt-distsql-handler ............. SKIPPED
[INFO] shardingsphere-encrypt-core ........................ SKIPPED
[INFO] shardingsphere-shadow-api .......................... SKIPPED
[INFO] shardingsphere-shadow-distsql-handler .............. SKIPPED
[INFO] shardingsphere-shadow-core ......................... SKIPPED
[INFO] shardingsphere-jdbc ................................ SKIPPED
[INFO] shardingsphere-jdbc-core ........................... SKIPPED
[INFO] shardingsphere-jdbc-spring ......................... SKIPPED
[INFO] shardingsphere-jdbc-spring-infra ................... SKIPPED
[INFO] shardingsphere-jdbc-spring-boot-starter-infra ...... SKIPPED
[INFO] shardingsphere-jdbc-spring-namespace-infra ......... SKIPPED
[INFO] shardingsphere-sharding-spring ..................... SKIPPED
[INFO] shardingsphere-sharding-spring-boot-starter ........ SKIPPED
[INFO] shardingsphere-shadow-spring ....................... SKIPPED
[INFO] shardingsphere-shadow-spring-boot-starter .......... SKIPPED
[INFO] shardingsphere-encrypt-spring ...................... SKIPPED
[INFO] shardingsphere-encrypt-spring-boot-starter ......... SKIPPED
[INFO] shardingsphere-readwrite-splitting-spring .......... SKIPPED
[INFO] shardingsphere-readwrite-splitting-spring-boot-starter SKIPPED
[INFO] shardingsphere-jdbc-transaction-spring ............. SKIPPED
[INFO] shardingsphere-jdbc-core-spring .................... SKIPPED
[INFO] shardingsphere-jdbc-core-spring-boot-starter ....... SKIPPED
[INFO] shardingsphere-sharding-spring-namespace ........... SKIPPED
[INFO] shardingsphere-readwrite-splitting-spring-namespace  SKIPPED
[INFO] shardingsphere-encrypt-spring-namespace ............ SKIPPED
[INFO] shardingsphere-shadow-spring-namespace ............. SKIPPED
[INFO] shardingsphere-jdbc-core-spring-namespace .......... SKIPPED
[INFO] shardingsphere-proxy ............................... SKIPPED
[INFO] shardingsphere-proxy-common ........................ SKIPPED
[INFO] shardingsphere-scaling-core ........................ SKIPPED
[INFO] shardingsphere-scaling-distsql-handler ............. SKIPPED
[INFO] shardingsphere-scaling-dialect ..................... SKIPPED
[INFO] shardingsphere-scaling-mysql ....................... SKIPPED
[INFO] shardingsphere-scaling-postgresql .................. SKIPPED
[INFO] shardingsphere-proxy-backend ....................... SKIPPED
[INFO] shardingsphere-proxy-frontend ...................... SKIPPED
[INFO] shardingsphere-proxy-frontend-spi .................. SKIPPED
[INFO] shardingsphere-proxy-frontend-core ................. SKIPPED
[INFO] shardingsphere-proxy-frontend-mysql ................ SKIPPED
[INFO] shardingsphere-proxy-frontend-postgresql ........... SKIPPED
[INFO] shardingsphere-proxy-frontend-opengauss ............ SKIPPED
[INFO] shardingsphere-scaling-opengauss ................... SKIPPED
[INFO] shardingsphere-proxy-bootstrap ..................... SKIPPED
[INFO] shardingsphere-db-discovery-spring ................. SKIPPED
[INFO] shardingsphere-db-discovery-spring-namespace ....... SKIPPED
[INFO] shardingsphere-db-discovery-spring-boot-starter .... SKIPPED
[INFO] shardingsphere-agent ............................... SKIPPED
[INFO] shardingsphere-agent-api ........................... SKIPPED
[INFO] shardingsphere-agent-core .......................... SKIPPED
[INFO] shardingsphere-agent-bootstrap ..................... SKIPPED
[INFO] shardingsphere-agent-plugins ....................... SKIPPED
[INFO] shardingsphere-agent-plugin-metrics ................ SKIPPED
[INFO] shardingsphere-agent-metrics-api ................... SKIPPED
[INFO] shardingsphere-agent-metrics-prometheus ............ SKIPPED
[INFO] shardingsphere-agent-plugin-tracing ................ SKIPPED
[INFO] shardingsphere-agent-tracing-test .................. SKIPPED
[INFO] shardingsphere-agent-tracing-jaeger ................ SKIPPED
[INFO] shardingsphere-agent-tracing-zipkin ................ SKIPPED
[INFO] shardingsphere-agent-tracing-opentracing ........... SKIPPED
[INFO] shardingsphere-agent-tracing-opentelemetry ......... SKIPPED
[INFO] shardingsphere-agent-plugin-logging ................ SKIPPED
[INFO] shardingsphere-agent-logging-base .................. SKIPPED
[INFO] shardingsphere-agent-distribution .................. SKIPPED
[INFO] shardingsphere-scaling-bootstrap ................... SKIPPED
[INFO] shardingsphere-integration-test .................... SKIPPED
[INFO] shardingsphere-integration-test-fixture ............ SKIPPED
[INFO] shardingsphere-integration-test-suite .............. SKIPPED
[INFO] shardingsphere-integration-driver-test ............. SKIPPED
[INFO] shardingsphere-integration-agent-test .............. SKIPPED
[INFO] shardingsphere-integration-agent-test-plugins ...... SKIPPED
[INFO] shardingsphere-integration-agent-test-common ....... SKIPPED
[INFO] shardingsphere-integration-agent-test-metrics ...... SKIPPED
[INFO] shardingsphere-integration-agent-test-zipkin ....... SKIPPED
[INFO] shardingsphere-integration-agent-test-jaeger ....... SKIPPED
[INFO] shardingsphere-integration-agent-test-opentelemetry  SKIPPED
[INFO] shardingsphere-integration-scaling-test ............ SKIPPED
[INFO] shardingsphere-integration-scaling-test-mysql ...... SKIPPED
[INFO] shardingsphere-rewrite-test ........................ SKIPPED
[INFO] shardingsphere-distribution ........................ SKIPPED
[INFO] shardingsphere-src-distribution .................... SKIPPED
[INFO] shardingsphere-jdbc-distribution ................... SKIPPED
[INFO] shardingsphere-proxy-distribution .................. SKIPPED
[INFO] shardingsphere-scaling-distribution ................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.454 s
[INFO] Finished at: 2021-09-15T14:22:53+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4:enforce (enforce-banned-dependencies) on project shardingsphere: Execution enforce-banned-dependencies of goal org.apache.maven.plugins:maven-enforcer-plugin:1.4:enforce failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-enforcer-plugin:1.4:enforce: java.lang.ExceptionInInitializerError: null
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.apache.maven.plugins:maven-enforcer-plugin:1.4
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/wuweijie/.m2/repository/org/apache/maven/plugins/maven-enforcer-plugin/1.4/maven-enforcer-plugin-1.4.jar
[ERROR] urls[1] = file:/Users/wuweijie/.m2/repository/backport-util-concurrent/backport-util-concurrent/3.1/backport-util-concurrent-3.1.jar
[ERROR] urls[2] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.11/plexus-interpolation-1.11.jar
[ERROR] urls[3] = file:/Users/wuweijie/.m2/repository/org/slf4j/slf4j-jdk14/1.5.6/slf4j-jdk14-1.5.6.jar
[ERROR] urls[4] = file:/Users/wuweijie/.m2/repository/org/slf4j/jcl-over-slf4j/1.5.6/jcl-over-slf4j-1.5.6.jar
[ERROR] urls[5] = file:/Users/wuweijie/.m2/repository/org/apache/maven/reporting/maven-reporting-api/2.2.1/maven-reporting-api-2.2.1.jar
[ERROR] urls[6] = file:/Users/wuweijie/.m2/repository/org/apache/maven/doxia/doxia-sink-api/1.1/doxia-sink-api-1.1.jar
[ERROR] urls[7] = file:/Users/wuweijie/.m2/repository/org/apache/maven/doxia/doxia-logging-api/1.1/doxia-logging-api-1.1.jar
[ERROR] urls[8] = file:/Users/wuweijie/.m2/repository/commons-cli/commons-cli/1.2/commons-cli-1.2.jar
[ERROR] urls[9] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-interactivity-api/1.0-alpha-4/plexus-interactivity-api-1.0-alpha-4.jar
[ERROR] urls[10] = file:/Users/wuweijie/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[11] = file:/Users/wuweijie/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[12] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.20/plexus-utils-3.0.20.jar
[ERROR] urls[13] = file:/Users/wuweijie/.m2/repository/commons-lang/commons-lang/2.3/commons-lang-2.3.jar
[ERROR] urls[14] = file:/Users/wuweijie/.m2/repository/org/apache/maven/enforcer/enforcer-api/1.4/enforcer-api-1.4.jar
[ERROR] urls[15] = file:/Users/wuweijie/.m2/repository/org/apache/maven/enforcer/enforcer-rules/1.4/enforcer-rules-1.4.jar
[ERROR] urls[16] = file:/Users/wuweijie/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/1.4/maven-common-artifact-filters-1.4.jar
[ERROR] urls[17] = file:/Users/wuweijie/.m2/repository/org/beanshell/bsh/2.0b4/bsh-2.0b4.jar
[ERROR] urls[18] = file:/Users/wuweijie/.m2/repository/org/apache/maven/shared/maven-dependency-tree/2.2/maven-dependency-tree-2.2.jar
[ERROR] urls[19] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[20] = file:/Users/wuweijie/.m2/repository/org/eclipse/aether/aether-util/0.9.0.M2/aether-util-0.9.0.M2.jar
[ERROR] urls[21] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-i18n/1.0-beta-6/plexus-i18n-1.0-beta-6.jar
[ERROR] urls[22] = file:/Users/wuweijie/.m2/repository/org/apache/maven/plugin-testing/maven-plugin-testing-harness/1.3/maven-plugin-testing-harness-1.3.jar
[ERROR] urls[23] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-archiver/2.2/plexus-archiver-2.2.jar
[ERROR] urls[24] = file:/Users/wuweijie/.m2/repository/org/codehaus/plexus/plexus-io/2.0.4/plexus-io-2.0.4.jar
[ERROR] urls[25] = file:/Users/wuweijie/.m2/repository/junit/junit/4.11/junit-4.11.jar
[ERROR] urls[26] = file:/Users/wuweijie/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[project>org.apache.shardingsphere:shardingsphere:5.0.0-RC1-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : begin 0, end 3, length 2
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException

```

```
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.06 s <<< FAILURE! - in org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest
[ERROR] assertFindCachedServices(org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest)  Time elapsed: 0.06 s  <<< ERROR!
java.lang.NoSuchFieldException: modifiers
	at org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest.cleanCache(OrderedServicesCacheTest.java:52)

[ERROR] assertNotFindCachedServices(org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NoSuchFieldException: modifiers
	at org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest.cleanCache(OrderedServicesCacheTest.java:52)

[INFO] Running org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest
[ERROR] Tests run: 3, Failures: 0, Errors: 3, Skipped: 0, Time elapsed: 0.003 s <<< FAILURE! - in org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest
[ERROR] assertGetRegisteredServicesByClass(org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest)  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NoSuchFieldException: modifiers
	at org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest.cleanCache(OrderedSPIRegistryTest.java:49)

[ERROR] assertGetRegisteredServicesFromCache(org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest)  Time elapsed: 0.002 s  <<< ERROR!
java.lang.NoSuchFieldException: modifiers
	at org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest.cleanCache(OrderedSPIRegistryTest.java:49)

[ERROR] assertGetRegisteredServices(org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NoSuchFieldException: modifiers
	at org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest.cleanCache(OrderedSPIRegistryTest.java:49)
```

Refer to https://bugs.openjdk.java.net/browse/JDK-8210522

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -80,6 +80,10 @@ jobs:
             version: 11,
             maven_args: "-Dmaven.javadoc.skip=true"
           }
+          - {
+            version: 17,
+            maven_args: "-Dmaven.javadoc.skip=true"
+          }
     steps:
       - name: Cache Maven Repos
         uses: actions/cache@v2

--- a/examples/pom.xml
+++ b/examples/pom.xml
@@ -51,7 +51,7 @@
         <h2.version>1.4.196</h2.version>
         <slf4j.version>1.7.7</slf4j.version>
         <logback.version>1.2.0</logback.version>
-        <lombok.version>1.18.16</lombok.version>
+        <lombok.version>1.18.20</lombok.version>
         <mybatis.version>3.5.1</mybatis.version>
         <mybatis-spring.version>2.0.1</mybatis-spring.version>
 

--- a/examples/pom.xml
+++ b/examples/pom.xml
@@ -71,8 +71,8 @@
         <apache-rat-plugin.version>0.12</apache-rat-plugin.version>
         <maven-compiler-plugin.version>3.3</maven-compiler-plugin.version>
         <maven-resources-plugin.version>2.7</maven-resources-plugin.version>
-        <maven-jar-plugin.version>2.6</maven-jar-plugin.version>
-        <maven-source-plugin.version>2.4</maven-source-plugin.version>
+        <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
+        <maven-source-plugin.version>3.2.1</maven-source-plugin.version>
         <freemarker.version>2.3.31</freemarker.version>
         
         <javadocExecutable>${java.home}/../bin/javadoc</javadocExecutable>

--- a/pom.xml
+++ b/pom.xml
@@ -100,7 +100,7 @@
         <opengauss.version>2.0.1-compatibility</opengauss.version>
         <mssql.version>6.1.7.jre8-preview</mssql.version>
         <mariadb-java-client.version>2.4.2</mariadb-java-client.version>
-        <bytebuddy.version>1.10.21</bytebuddy.version>
+        <bytebuddy.version>1.11.18</bytebuddy.version>
         
         <elasticjob.version>3.0.0-beta</elasticjob.version>
         

--- a/pom.xml
+++ b/pom.xml
@@ -100,7 +100,7 @@
         <opengauss.version>2.0.1-compatibility</opengauss.version>
         <mssql.version>6.1.7.jre8-preview</mssql.version>
         <mariadb-java-client.version>2.4.2</mariadb-java-client.version>
-        <bytebuddy.version>1.10.16</bytebuddy.version>
+        <bytebuddy.version>1.10.21</bytebuddy.version>
         
         <elasticjob.version>3.0.0-beta</elasticjob.version>
         
@@ -132,7 +132,7 @@
         <lifecycle-mapping.version>1.0.0</lifecycle-mapping.version>
         <coveralls-maven-plugin.version>4.3.0</coveralls-maven-plugin.version>
         <cobertura-maven-plugin.version>2.7</cobertura-maven-plugin.version>
-        <jacoco-maven-plugin.version>0.8.5</jacoco-maven-plugin.version>
+        <jacoco-maven-plugin.version>0.8.7</jacoco-maven-plugin.version>
         <findbugs-maven-plugin.version>3.0.2</findbugs-maven-plugin.version>
         <maven-checkstyle-plugin.version>3.1.0</maven-checkstyle-plugin.version>
         <maven-pmd-plugin.version>3.5</maven-pmd-plugin.version>

--- a/pom.xml
+++ b/pom.xml
@@ -81,7 +81,7 @@
         <curator.version>5.1.0</curator.version>
         <jetcd.version>0.5.0</jetcd.version>
         
-        <lombok.version>1.18.16</lombok.version>
+        <lombok.version>1.18.20</lombok.version>
         
         <springframework.version>[4.3.6.RELEASE,5.0.0.M1)</springframework.version>
         <spring-boot.version>[1.5.20.RELEASE,2.0.0.M1)</spring-boot.version>
@@ -118,16 +118,16 @@
         <maven-compiler-plugin.version>3.8.0</maven-compiler-plugin.version>
         <maven-assembly-plugin.version>3.1.0</maven-assembly-plugin.version>
         <maven-resources-plugin.version>2.7</maven-resources-plugin.version>
-        <maven-jar-plugin.version>2.6</maven-jar-plugin.version>
+        <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
         <maven-surefire-plugin.version>2.18.1</maven-surefire-plugin.version>
         <maven-deploy-plugin.version>2.8.2</maven-deploy-plugin.version>
         <maven-release-plugin.version>2.5.3</maven-release-plugin.version>
         <maven-site-plugin.version>3.4</maven-site-plugin.version>
-        <maven-enforcer-plugin.version>1.4</maven-enforcer-plugin.version>
+        <maven-enforcer-plugin.version>3.0.0</maven-enforcer-plugin.version>
         <maven-project-info-reports-plugin.version>2.8</maven-project-info-reports-plugin.version>
         <maven-plugin-plugin.version>3.4</maven-plugin-plugin.version>
         <maven-javadoc-plugin.version>3.3.0</maven-javadoc-plugin.version>
-        <maven-source-plugin.version>2.4</maven-source-plugin.version>
+        <maven-source-plugin.version>3.2.1</maven-source-plugin.version>
         <maven-jxr-plugin.version>2.5</maven-jxr-plugin.version>
         <lifecycle-mapping.version>1.0.0</lifecycle-mapping.version>
         <coveralls-maven-plugin.version>4.3.0</coveralls-maven-plugin.version>

--- a/shardingsphere-infra/shardingsphere-infra-executor/pom.xml
+++ b/shardingsphere-infra/shardingsphere-infra-executor/pom.xml
@@ -74,6 +74,7 @@
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
                             <excludes>
+                                <!-- TODO Consider refactoring tests which changed Field's modifiers -->
                                 <exclude>org.apache.shardingsphere.infra.executor.sql.log.SQLLoggerTest</exclude>
                             </excludes>
                         </configuration>

--- a/shardingsphere-infra/shardingsphere-infra-executor/pom.xml
+++ b/shardingsphere-infra/shardingsphere-infra-executor/pom.xml
@@ -60,4 +60,26 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+    
+    <profiles>
+        <profile>
+            <id>jdk12+</id>
+            <activation>
+                <jdk>[12,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>org.apache.shardingsphere.infra.executor.sql.log.SQLLoggerTest</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-opengauss/pom.xml
+++ b/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-opengauss/pom.xml
@@ -40,4 +40,26 @@
             <version>${project.version}</version>
         </dependency>
     </dependencies>
+    
+    <profiles>
+        <profile>
+            <id>jdk12+</id>
+            <activation>
+                <jdk>[12,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>org.apache.shardingsphere.proxy.frontend.opengauss.OpenGaussFrontendEngineTest</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-opengauss/pom.xml
+++ b/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-opengauss/pom.xml
@@ -54,6 +54,7 @@
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
                             <excludes>
+                                <!-- TODO Consider refactoring tests which changed Field's modifiers -->
                                 <exclude>org.apache.shardingsphere.proxy.frontend.opengauss.OpenGaussFrontendEngineTest</exclude>
                             </excludes>
                         </configuration>

--- a/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/pom.xml
+++ b/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/pom.xml
@@ -67,6 +67,7 @@
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
                             <excludes>
+                                <!-- TODO Consider refactoring tests which changed Field's modifiers -->
                                 <exclude>org.apache.shardingsphere.proxy.frontend.postgresql.command.query.text.PostgreSQLComQueryExecutorTest</exclude>
                             </excludes>
                         </configuration>

--- a/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/pom.xml
+++ b/shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-postgresql/pom.xml
@@ -53,4 +53,26 @@
             <artifactId>netty-all</artifactId>
         </dependency>
     </dependencies>
+    
+    <profiles>
+        <profile>
+            <id>jdk12+</id>
+            <activation>
+                <jdk>[12,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>org.apache.shardingsphere.proxy.frontend.postgresql.command.query.text.PostgreSQLComQueryExecutorTest</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/shardingsphere-spi/pom.xml
+++ b/shardingsphere-spi/pom.xml
@@ -41,6 +41,7 @@
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
                             <excludes>
+                                <!-- TODO Consider refactoring tests which changed Field's modifiers -->
                                 <exclude>org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest</exclude>
                                 <exclude>org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest</exclude>
                             </excludes>

--- a/shardingsphere-spi/pom.xml
+++ b/shardingsphere-spi/pom.xml
@@ -27,4 +27,27 @@
     </parent>
     <artifactId>shardingsphere-spi</artifactId>
     <name>${project.artifactId}</name>
+    
+    <profiles>
+        <profile>
+            <id>jdk12+</id>
+            <activation>
+                <jdk>[12,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>org.apache.shardingsphere.spi.ordered.cache.OrderedServicesCacheTest</exclude>
+                                <exclude>org.apache.shardingsphere.spi.ordered.OrderedSPIRegistryTest</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-engine/pom.xml
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-engine/pom.xml
@@ -47,4 +47,26 @@
             <artifactId>antlr4-runtime</artifactId>
         </dependency>
     </dependencies>
+    
+    <profiles>
+        <profile>
+            <id>jdk12+</id>
+            <activation>
+                <jdk>[12,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>org.apache.shardingsphere.sql.parser.api.SQLParserEngineTest</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
 </project>

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-engine/pom.xml
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-engine/pom.xml
@@ -61,6 +61,7 @@
                         <artifactId>maven-surefire-plugin</artifactId>
                         <configuration>
                             <excludes>
+                                <!-- TODO Consider refactoring tests which changed Field's modifiers -->
                                 <exclude>org.apache.shardingsphere.sql.parser.api.SQLParserEngineTest</exclude>
                             </excludes>
                         </configuration>

